### DISCUSSION TITLE
GitHub Actions Importer - [GA]

### DISCUSSION THREAD

**Discussion by tuves** [community/community#48913] [Announcements]
## New Release: GitHub Actions Importer - [GA]

We’re excited to announce the general availability of GitHub Actions Importer. GitHub Actions Importer helps you plan, forecast, and automate migrations from Azure DevOps, CircleCI, GitLab, Jenkins, and Travis CI to GitHub Actions. This product is an extension of the official [GitHub CLI](https://cli.github.com/) and is available for free to any GitHub user starting today.
Migrating to a new CI/CD tool is often a manual and time-intensive endeavor. This is more pronounced for users that have a large and sophisticated CI/CD infrastructure—for instance, some of our largest customers have more than 15K pipelines to migrate. Additionally, manual migrations introduce the potential for regressions in newly created pipelines. GitHub Actions Importer is designed to help plan and automate a large portion of the migration to provide teams with the speed and confidence to continue delivering software efficiently.

### Why GitHub Actions? 
GitHub Actions provides access to powerful, native CI/CD capabilities right next to your code hosted in GitHub. Actions is simple to enable and painless to maintain within your current GitHub workflow as there’s no longer a need to install, integrate, and maintain a third-party tool. GitHub Actions, along with other GitHub Enterprise offerings, [improves developer productivity by 22%](https://resources.github.com/forrester/?utm_source=&utm_medium=&utm_campaign=2020q3-site-ww-ForresterReport&utm_content=Blog) and reduces time spent managing tools and code infrastructure by 75%. Additionally, thousands of pre-built actions from trusted, verified partners and the OSS community make it easier than ever to deliver your software without ever leaving the GitHub interface.

### How it works
GitHub Actions Importer uses a phased approach to simplify the migration process.

1. Planning - In this phase, you analyze your existing CI/CD usage to build a roadmap for your migration.
2. Testing - Here you conduct dry-run migrations to validate that the converted workflows function the same as existing pipelines. GitHub Actions Importer supports unlimited iterations in this step to ensure any custom behavior is accurately encapsulated in your new GitHub Actions workflows.
3. Migration - In this last phase, GitHub Actions Importer generates validated workflows and opens pull requests to add them to your GitHub repository. To finalize your migration, you should plan to review these workflows and migrate those constructs that could not be migrated automatically.

### Getting started
Head over to the GitHub [documentation](https://gh.io/actions-importer-documentation) to learn how to get started with GitHub Actions Importer today. Check out the [Demo Video](https://youtu.be/WqiGP6h4fa0).

### Looking ahead
Over the coming months, we will continue to enhance and update GitHub Actions Importer to include support for additional CI/CD tools. Stay tuned to the GitHub [roadmap](https://github.com/orgs/github/projects/4247) for information about what’s coming to GitHub Actions Importer. As always, we would love to hear from you. You can provide feedback by [posting here](https://gh.io/actions-importer-feedback).

**Comment by novium2** [comment]
Retiring this post as it is over 6 months old. If you have any questions, consider starting a [new post](https://github.com/orgs/community/discussions/new/choose).
